### ISSUE TITLE
When upgrading to FastExpressionCompiler 1.5 exception is thrown on session.Store

### ISSUE THREAD

**Issue by wastaz** [JasperFx/marten#921]
I upgraded my FastExpressionCompiler dependency from 1.4 to 1.5, this causes the following code to crash with the exception given below.

Workaround: Downgrade FastExpressionCompiler to 1.4

A possible quickfix here if may be to just limit Martens dependency on FastExpressionCompiler to not pull in a version higher than 1.4?

```
using System;
using Marten;

namespace MartenExceptionRepro
{
    public class TestDocument
    {
        public Guid Id { get; set; }
        public string Foo { get; set; }
        public DateTime Date { get; set; } 
    }
    
    public class Program
    {
        public static void Main(string[] args)
        {
            var docStore = DocumentStore.For(x =>
            {
                x.Connection("host=localhost;port=5432;database=marten;username=postgres;password=postgres;");
            });

            using (var session = docStore.LightweightSession())
            {
                session.Store(new TestDocument
                {
                    Id = Guid.NewGuid(),
                    Foo = "Blah",
                    Date = DateTime.UtcNow,
                });
                session.SaveChanges();
            }
        }
    }
}
```

Exception:
```
{System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.ArgumentException: Cannot bind to the target method because its signature or security transparency is not compatible with that of the delegate type.
   at System.Delegate.CreateDelegateNoSecurityCheck(Type type, Object target, RuntimeMethodHandle method)
   at System.Reflection.Emit.DynamicMethod.CreateDelegate(Type delegateType)
   at FastExpressionCompiler.ExpressionCompiler.TryCompileStaticDelegate(Type delegateType, Type[] paramTypes, Type returnType, Object exprObj, ExpressionType exprNodeType, Type exprType, IList`1 paramExprs)
   at FastExpressionCompiler.ExpressionCompiler.TryCompile(ClosureInfo& closureInfo, Type delegateType, Type[] paramTypes, Type returnType, Object exprObj, ExpressionType exprNodeType, Type exprType, IList`1 paramExprs, Boolean isNestedLambda)
   at FastExpressionCompiler.ExpressionCompiler.TryCompile[TDelegate](Expression bodyExpr, IList`1 paramExprs, Type[] paramTypes, Type returnType)
   at FastExpressionCompiler.ExpressionCompiler.TryCompile[TDelegate](LambdaExpression lambdaExpr)
   at FastExpressionCompiler.ExpressionCompiler.Compile[TDelegate](LambdaExpression lambdaExpr)
   at Marten.Schema.DocumentStorage`1.buildSprocWriter(DocumentMapping mapping)
   at Marten.Schema.DocumentStorage`1..ctor(ISerializer serializer, DocumentMapping mapping)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.RuntimeType.CreateInstanceImpl(BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes, StackCrawlMark& stackMark)
   at System.Activator.CreateInstance(Type type, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes)
   at System.Activator.CreateInstance(Type type, Object[] args)
   at Marten.Schema.DocumentMapping.Marten.Schema.IDocumentMapping.BuildStorage(StoreOptions options)
   at Marten.Storage.StorageFeatures.<>c__DisplayClass21_0.<StorageFor>b__0(Type type)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Marten.Storage.StorageFeatures.StorageFor(Type documentType)
   at Marten.Storage.Tenant.StorageFor(Type documentType)
   at Marten.Storage.LightweightTenant.StorageFor(Type documentType)
   at Marten.DocumentSession.Store[T](T[] entities)
   at MartenExceptionRepro.Program.Main(String[] args) in C:\code\temp\MartenExceptionRepro\Program.cs:line 24}
```

**Comment by wastaz** [comment]
@jeremydmiller Is fixing the nuget dependency on FastExpressionCompiler an accepted fix for this issue or do you want a better fix for this? I may be able to send a PR for this, but I dont want to do something that you dont think is a good solution :)

**Comment by jeremydmiller** [comment]
@wastaz I say we copy the FastExpressionCompiler code (it's one big file as I recall) into Marten, then make it internal and remove the nuget dependency. That's what I did for StructureMap and what the FEC author does himself in other cases.

**Comment by wastaz** [comment]
@jeremydmiller That sounds like a good option, the less dependencies the less potential worries for nuget version hell :)

**Comment by wastaz** [comment]
Ended up falling into this pit again today when FastExpressionCompiler was accidentially upgraded. It would be really nice to have this be a part of the next release of Marten so I dont end up spending more time on this the next time I am a klutz :D

Do you think that you will have the time to do this fix @jeremydmiller before next Marten release or do you want me to send in a PR for it? :)

**Comment by jeremydmiller** [comment]
The open issue count is bonkers right now, so I'll happily take a PR. To inline it as source code, right?

**Comment by wastaz** [comment]
PR #968 should fix this.

I have inlined the same version as is used in StructureMap as source code.
I have tried to find the "latest" version that works, but I ended up giving up on that quest. I know that 1.5 and 1.6 does not work, whereas 1.4 does. But which commit in FastExpressionCompiler is the commit that 1.4 is built off....I have zero clues to be honest. This version is probably quite a bit older than 1.4, but since structuremap is used so much I figured that we can probably feel pretty safe using this anyway, also all tests seems to run fine on my computer so... :)

I really would like to find the underlying cause for this, but I have so far not been able to figure it out (I actually started working on this PR about an hour before you replied ;) )
